test/unit/lib/migration-chunks/validation/field/change-id.spec.js

```diff
@@ -0,0 +1,219 @@
+      migration.editContentType('movie').changeFieldId('title', 'newTitle');
+      book.editField('newTitle').name('new Title');
+      book.changeFieldId('newTitle', 'title');
+      book.editField('title').type('Symbol');
+    });
+    const plan = stripCallsites(migrationPlan(steps));
+    const contentTypes = [{
+      sys: { id: 'book' },
+      fields: [{
+        id: 'title'
+      }]
+    }, {
+      sys: { id: 'movie' },
+      fields: [{
+        id: 'title'
+      }]
+    }];
+
+    const validationErrors = validatePlan(plan, contentTypes);
+    expect(validationErrors).to.eql([]);
+  }));
+
+  it('does not return errors when switching the id for a field and then creating a field with the same id later on', Bluebird.coroutine(function * () {
+    const steps = yield migrationSteps(function up (migration) {
+      const book = migration.editContentType('book');
+      book.changeFieldId('title', 'newTitle');
+      book.createField('title', {
+        name: 'titulo',
+        type: 'Symbol'
+      });
+    });
+    const plan = stripCallsites(migrationPlan(steps));
+    const contentTypes = [{
+      sys: { id: 'book' },
+      fields: [{
+        id: 'title'
+      }]
+    }];
+
+    const validationErrors = validatePlan(plan, contentTypes);
+    expect(validationErrors).to.eql([]);
+  }));
+
+  it('does return an error when switching the id of a field to an already existing one', Bluebird.coroutine(function * () {
+    const steps = yield migrationSteps(function up (migration) {
+      const book = migration.editContentType('book');
+      book.changeFieldId('title', 'pages');
+    });
+    const plan = stripCallsites(migrationPlan(steps));
+    const contentTypes = [{
+      sys: { id: 'book' },
+      fields: [{
+        id: 'title'
+      }, {
+        id: 'pages'
+      }]
+    }];
+
+    const validationErrors = validatePlan(plan, contentTypes);
+    expect(validationErrors).to.eql([
+      {
+        type: 'InvalidAction',
+        message: 'Cannot rename field "title" to "pages" because a field with this ID already exists on content type "book".',
+        details: {
+          step: {
+            'type': 'field/rename',
+            'meta': {
+              'contentTypeInstanceId': 'contentType/book/0',
+              'fieldInstanceId': 'fields/title/0'
+            },
+            'payload': {
+              'contentTypeId': 'book',
+              'fieldId': 'title',
+              'props': {
+                'newId': 'pages'
+              }
+            }
+          }
+        }
+      }
+    ]);
+  }));
+
+  it('does return an error when switching the id of 2 fields to the same id', Bluebird.coroutine(function * () {
+    const steps = yield migrationSteps(function up (migration) {
+      const book = migration.editContentType('book');
+      book.changeFieldId('title', 'newTitle');
+      book.changeFieldId('pages', 'title');
+      book.changeFieldId('newTitle', 'title');
```

Comment:
Maybe not rename `title -> newTitle` first, as it's then a "circular renaming" again.